contributing/sdks.md

```diff
@@ -0,0 +1,577 @@
+
+**Key Point**: SDK generation is a **manual process** that requires explicit workflow triggering and version management. This is intentional to maintain control over releases.
+
+## Prerequisites
+
+### Required Tools
+- Git
+- GitHub account with write access to the Outpost repository
+- For local testing:
+  - **Go SDK**: Go 1.21+
+  - **Python SDK**: Python 3.9+, pip
+  - **TypeScript SDK**: Node.js 18+, npm
+
+### Required Access & Secrets
+The following secrets must be configured in the repository (for maintainers):
+- `SPEAKEASY_API_KEY`: For SDK generation
+- `NPM_TOKEN`: For publishing TypeScript SDK to npm
+- `PYPI_API_TOKEN`: For publishing Python SDK to PyPI
+- GitHub token (automatic): For publishing Go SDK
+
+## SDK Architecture
+
+Outpost maintains three official SDKs, each in its own directory within the `sdks/` folder:
+
+### 1. Go SDK (`sdks/outpost-go/`)
+- **Location**: `sdks/outpost-go/`
+- **Package**: `github.com/hookdeck/outpost-go`
+- **Published**: Automatically via Git tags
+- **Generation Workflow**: `sdk_generation_outpost_go.yaml`
+- **Publish Workflow**: `sdk_publish_outpost_go.yaml`
+
+### 2. Python SDK (`sdks/outpost-python/`)
+- **Location**: `sdks/outpost-python/`
+- **Package**: `outpost-python` on PyPI
+- **Published**: Automatically to PyPI on merge
+- **Generation Workflow**: `sdk_generation_outpost_python.yaml`
+- **Publish Workflow**: `sdk_publish_outpost_python.yaml`
+
+### 3. TypeScript SDK (`sdks/outpost-typescript/`)
+- **Location**: `sdks/outpost-typescript/`
+- **Package**: `@hookdeck/outpost` on npm (also `@hookdeck/outpost-sdk`)
+- **Published**: Automatically to npm on merge
+- **Generation Workflow**: `sdk_generation_outpost_ts.yaml`
+- **Publish Workflow**: `sdk_publish_outpost_ts.yaml`
+
+Each SDK is maintained as a sub-project within this repository and generated from the shared [OpenAPI specification](../docs/apis/openapi.yaml).
+
+## Updating the OpenAPI Specification
+
+All SDK changes begin with updating the OpenAPI specification:
+
+1. **Edit the specification**:
+   ```bash
+   # The OpenAPI spec is located at:
+   docs/apis/openapi.yaml
+   ```
+
+2. **Make your changes**:
+   - Add new endpoints, modify existing ones, or update schemas
+   - Follow OpenAPI 3.0 standards
+   - Ensure all changes are backward compatible when possible
+   - Add descriptions and examples for better SDK documentation
+
+3. **Validate the specification**:
+   ```bash
+   # Use Speakeasy CLI to validate (if installed)
+   speakeasy validate -s docs/apis/openapi.yaml
+   
+   # Or use the spec-sdk-tests validation
+   cd spec-sdk-tests
+   npm run validate:spec
+   ```
+
+4. **Commit and push**:
+   ```bash
+   git add docs/apis/openapi.yaml
+   git commit -m "Update OpenAPI spec: <description>"
```

Comment:
I don't think this commit message conform to conventional commit convention which we follow